### DISCUSSION TITLE
Add a method for user to delete one of headers from a request

### DISCUSSION THREAD

**Discussion by NewUserHa** [encode/httpx#3037] [Potential Issue]
Currently, users can't make a header not to send out in `httpx`, in practice, like `refer:` `origin:` `X-Requested-With:` is not needed for some requests.

it should allow:
```
import httpx
s = httpx.AsyncClient(headers={"refer": "example.com", "origin": "example.com"})
r = await s.get("http://example.com", headers={"origin": None})
```
to archive the same behavior of `requests` (handle with merge_setting() from prepare_request() of Session() in seesion.py)

Perhaps should make `normalize_header_value()` of `Headers()` in '_models.py' to support `None` to fix
since `_merge_headers()` of `build_request` instance both 'session.headers' and 'headers' argument to `Headers()` before`update()`?

(also here should avoid re-instance the `self.headers` multiple times since it already is `Headers()`?)

**Comment by scottzach1** [comment]
Related to #152

**Comment by injust** [comment]
For my use case, I need to remove the `Accept` header that is automatically added in the `BaseClient.headers` setter.

Right now, I'm doing this with:

```python
httpx_client = httpx.AsyncClient()
del httpx_client.headers["Accept"]
```

Instead, it would be nice if this were possible:

```python
httpx_client = httpx.AsyncClient(headers={"Accept": None})
```

**Comment by technillogue** [comment]
Hi, I have a similar use-case

```
def __init__(self):
    self.headers = {"Authorization": "...", "X-Worker-Version": "...", "Other-Secret-Value": "...", ...}
    self.client = httpx.AsyncClient(headers=self.headers)

# many methods that require secret headers

async def get_untrusted_resource(self, url):
    await self.client.get(url, headers={}) # doesn't work
    await self.client.get(url, headers={k: None for k in self.headers}) # Header value must be str or bytes, not <class 'NoneType'>
    req = self.client.build_request("GET", url)
    req.headers = {}
    await self.client.send(req) # 'dict' object has no attribute 'raw'
    req.headers = httpx.Headers()
    await self.client.send(req) # Missing mandatory Host: header
    req.headers = httpx.Headers({k: v for k, v in req.headers.items() if k not in self.headers})
    await self.client.send(req) # this works but seems uglier 
```

In this case, I some uses of the client must not have the headers configured on the client, but I still want to use the same connection pool.
